src/unitquaternion.jl

```diff
@@ -305,22 +312,10 @@ function Base.:*(q::UnitQuaternion, r::StaticVector)  # must be StaticVector to
     (w^2 - v'v)*r + 2*v*(v'r) + 2*w*cross(v,r)
 end
 
-"""
-    (*)(q::UnitQuaternion, w::Real)
-
-Scalar multiplication of a quaternion. Breaks unit norm.
-"""
-function (*)(q::Q, w::Real) where Q<:UnitQuaternion
-    return Q(q.w*w, q.x*w, q.y*w, q.z*w, false)
-end
-(*)(w::Real, q::UnitQuaternion) = q*w
-
+(\)(q1::UnitQuaternion, q2::UnitQuaternion) = inv(q1)*q2  # Equivalent to inv(q1)*q2
+(/)(q1::UnitQuaternion, q2::UnitQuaternion) = q1*inv(q2)  # Equivalent to q1*inv(q2)
 
-
-(\)(q1::UnitQuaternion, q2::UnitQuaternion) = conj(q1)*q2  # Equivalent to inv(q1)*q2
-(/)(q1::UnitQuaternion, q2::UnitQuaternion) = q1*conj(q2)  # Equivalent to q1*inv(q2)
-
-(\)(q::UnitQuaternion, r::SVector{3}) = conj(q)*r          # Equivalent to inv(q)*r
+(\)(q::UnitQuaternion, r::SVector{3}) = inv(q)*r          # Equivalent to inv(q)*r
```

Comment:
Aren't these defined for all rotations? (I.e. can they be deleted instead?)

Reply:
With original definition:
```julia
julia> @benchmark q\SVector(1,2,3)  # q isa UnitQuaternion
BenchmarkTools.Trial: 10000 samples with 996 evaluations.
 Range (min … max):  25.933 ns …  1.724 μs  ┊ GC (min … max): 0.00% … 97.61%
 Time  (median):     26.546 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   28.727 ns ± 42.051 ns  ┊ GC (mean ± σ):  3.85% ±  2.59%

  ▂▆█▇▄▂                        ▁▂▂▂▂▂▁▁                      ▁
  ██████▇▆▇▆▆▆▅▅▅▅▃▅▅▆▆▆▅▅▅▆▇▇▇██████████▇▆▆▆▅▄▄▆▆▆▅▅▆▅▄▅▄▄▄▆ █
  25.9 ns      Histogram: log(frequency) by time      38.8 ns <

 Memory estimate: 32 bytes, allocs estimate: 1.
```

If Removing the definition:
```julia
julia> @benchmark q\SVector(1,2,3)  # q isa UnitQuaternion
BenchmarkTools.Trial: 10000 samples with 780 evaluations.
 Range (min … max):  160.083 ns …  2.153 μs  ┊ GC (min … max): 0.00% … 92.03%
 Time  (median):     164.683 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   167.640 ns ± 44.395 ns  ┊ GC (mean ± σ):  0.59% ±  2.06%

        ▂▃█▁▁                                                   
  ▂▂▁▁▃▇█████▇▇▇▄▄▃▃▃▃▃▃▃▄▄▄▄▃▃▃▃▃▃▃▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂ ▃
  160 ns          Histogram: frequency by time          186 ns <

 Memory estimate: 32 bytes, allocs estimate: 1.
```

This performance deterioration is because `(\).(::Rotation, ::SVector{3})` is not defined.

Reply:
So, we need to update the `\` definition here. With this update, we'll get more performance with other rotation types.

```julia
julia> using Rotations, StaticArrays, BenchmarkTools

julia> r = RotX(2.3)
3×3 RotX{Float64} with indices SOneTo(3)×SOneTo(3)(2.3):
 1.0   0.0        0.0
 0.0  -0.666276  -0.745705
 0.0   0.745705  -0.666276

julia> @benchmark r\SVector(1,2,3)
BenchmarkTools.Trial: 10000 samples with 861 evaluations.
 Range (min … max):  138.636 ns …  1.605 μs  ┊ GC (min … max): 0.00% … 90.34%
 Time  (median):     154.602 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   156.485 ns ± 31.911 ns  ┊ GC (mean ± σ):  0.45% ±  2.01%

                             ▂▇▇▅█▆▄▃▄                          
  ▂▁▁▁▁▂▁▂▂▁▂▂▂▂▂▂▂▂▃▂▂▃▃▄▄▄▆█████████▇█▄▃▄▃▄▃▄▄▄▄▅▅▅▄▃▃▂▂▂▂▂▂ ▄
  139 ns          Histogram: frequency by time          168 ns <

 Memory estimate: 32 bytes, allocs estimate: 1.

julia> Base.:\(r::Rotation, v::SVector{3}) = inv(r)*v

julia> @benchmark r\SVector(1,2,3)
BenchmarkTools.Trial: 10000 samples with 994 evaluations.
 Range (min … max):  29.926 ns …  1.726 μs  ┊ GC (min … max): 0.00% … 97.98%
 Time  (median):     30.742 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   32.839 ns ± 44.139 ns  ┊ GC (mean ± σ):  3.54% ±  2.59%

  ▃███▇▅▃▂                     ▁▂▂▂▂▂▂▂▁                      ▂
  ██████████▇▇▇▆▇█▆▆▆▆▅▆▇▆▆▇▇▇███████████▇▆▆▅▅▅▆▅▅▆▅▄▄▄▅▅▅▅▅▆ █
  29.9 ns      Histogram: log(frequency) by time      43.2 ns <

 Memory estimate: 32 bytes, allocs estimate: 1.
```

Reply:
I've added the method. (https://github.com/JuliaGeometry/Rotations.jl/pull/175/commits/9647c8476a848199b124dc20807dc956bc9f1102)